### PR TITLE
Remove directories from the cache

### PR DESCRIPTION
**Pull request by orta**
**What's the problem this PR addresses?**

`yarn install` fails when deleting a directory inside the cache dir. 

[Example](https://github.com/microsoft/TypeScript-Website/pull/757/checks?check_run_id=869050897#step:7:102)

**How did you fix it?**

Check whether the cache key being deleted is a file before deleting 

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [ ] I have verified that all automated PR checks pass.

I'll let CI do it

### CODE REVIEW

File: packages/yarnpkg-core/sources/Project.ts [hunk 1]
```diff
@@ -1679,7 +1679,12 @@ export class Project {
         report.reportError(MessageName.IMMUTABLE_CACHE, `${this.configuration.format(ppath.basename(entryPath), `magenta`)} appears to be unused and would marked for deletion, but the cache is immutable`);
       } else {
         report.reportInfo(MessageName.UNUSED_CACHE_ENTRY, `${this.configuration.format(ppath.basename(entryPath), `magenta`)} appears to be unused - removing`);
-        await xfs.unlinkPromise(entryPath);
```
Comment (merceyz) [inline packages/yarnpkg-core/sources/Project.ts:1682]: You can use `xfs.removePromise` which handles both

Comment (orta) [reply packages/yarnpkg-core/sources/Project.ts:1682]: Ace

**Comment by arcanis** [comment]
Sounds good if the tests are happy - how come you get a `.tmp` folder, though? 🤔

**Comment by orta** [comment]
It sounds like something I would have done while debugging, but its on CI, which is weird. I may have accidentally added it, it got put into the CI cache and then later I reverted the gitignore... Maybe, but it feels unlikely.

The repo doesn't have the cache folder at all: https://github.com/microsoft/TypeScript-Website/tree/v2/.yarn

**Pull request merged**